compiler/sigmatch.nim

```diff
@@ -985,6 +985,11 @@ proc inferStaticsInRange(c: var TCandidate,
         return isGeneric
       else:
         return isNone
+    if concrete.kind == tyRange:
+      var err = false
+      let lenOrd = getOrdValueAux(concrete.n[1], err)
+      if err:
+        return isNone
```

Comment:
I think an alternative to this would be to change the code below:

```nim
if fRange.rangeHasUnresolvedStatic:
  return inferStaticsInRange(c, fRange, a)
```

to:

```nim
if fRange.rangeHasUnresolvedStatic:
  if not aRange.rangeHasUnresolvedStatic:
    return inferStaticsInRange(c, fRange, a)
  else:
    return isNone
```

The issue with the ord value failing is that `f` and `a` both have unresolved statics (which is supposed to happen, `checkGeneric` in `cmpCandidates` compares generic overloads to each other), and the `inferStaticsInRange` tries to evaluate unresolved statics in `a` to assign to the unresolved statics in `f` and crashes because the unresolved statics don't have values. In this case calling `inferStaticsInRange` at all doesn't make sense, we can deal with it early on by checking if `a` has unresolved statics.

This change compiles the test but idk if it'll break anything in CI in general.

Reply:
As per my comment above the test cases that bring us to this point are avoided by the extra call to `typeRel` from the "try this" push. If that is a good sanity check then maybe it's worth trying it out anyway. Maybe run the CI with the 
```nim
proc view(a: Limbs) = discard
proc view(a: var Limbs) = discard
```
overload scenario on top of `devel` to see if it's worth merging before this PR?